src/github/taskDashboardWebviewProvider.ts

```diff
@@ -0,0 +1,1180 @@
+				}));
+			}
+		}));
+
+		// Also listen for when repositories are loaded within existing folder managers
+		for (const folderManager of this._repositoriesManager.folderManagers) {
+			this._register(folderManager.onDidLoadRepositories(() => {
+				this.updateDashboard();
+			}));
+		}
+	}
+
+	private registerFolderManagerBranchListeners(folderManager: FolderRepositoryManager): void {
+		// Listen for repository HEAD changes (branch changes)
+		this._register(folderManager.repository.state.onDidChange(() => {
+			// Debounce the update to avoid too frequent refreshes
+			if (this._branchChangeTimeout) {
+				clearTimeout(this._branchChangeTimeout);
+			}
+			this._branchChangeTimeout = setTimeout(() => {
+				this.updateDashboard();
+			}, 300); // 300ms debounce
+		}));
+	}
+
+	private _branchChangeTimeout: NodeJS.Timeout | undefined;
+
+	/**
+	 * Handles local task submission with issue support - creates branches and formats prompts
+	 */
+	private async handleLocalTaskWithIssueSupport(query: string): Promise<void> {
+		const references = extractIssueReferences(query);
+
+		if (references.length > 0) {
+			const firstRef = references[0];
+			const issueNumber = firstRef.number;
+
+			try {
+				await this._taskManager.handleLocalTaskForIssue(issueNumber, firstRef);
+			} catch (error) {
+				Logger.error(`Failed to handle local task with issue support: ${error}`, DashboardWebviewProvider.ID);
+				vscode.window.showErrorMessage('Failed to set up local task branch.');
+			}
+		} else {
+			await this.setupLocalWorkflow(query);
+		}
+	}
+
+	private getHtmlForWebview(): string {
+		const nonce = getNonce();
+		const uri = vscode.Uri.joinPath(this._context.extensionUri, 'dist', 'webview-dashboard.js');
+		const codiconsUri = this._webview!.asWebviewUri(vscode.Uri.joinPath(this._context.extensionUri, 'node_modules', '@vscode/codicons', 'dist', 'codicon.css'));
+
+		return `<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta charset="UTF-8">
+		<meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src vscode-resource: https: data:; script-src 'nonce-${nonce}' 'unsafe-eval' vscode-resource:; style-src vscode-resource: 'unsafe-inline' http: https: data:; font-src vscode-resource: data: 'self' https://*.vscode-cdn.net; worker-src 'self' blob: data:; connect-src 'self' https:;">
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<link href="${codiconsUri}" rel="stylesheet" />
+		<title>GitHub Dashboard</title>
+	</head>
+	<body>
+		<div id="app"></div>
+		<script nonce="${nonce}" src="${this._webview!.asWebviewUri(uri).toString()}"></script>
+	</body>
+</html>`;
+	}
+}
+
+
+/**
+ * Extracts issue references from text (e.g., #123, owner/repo#456)
+ */
+function extractIssueReferences(text: string): Array<IssueReference> {
+	const out: IssueReference[] = [];
+
+	// Match full repository issue references (owner/repo#123)
+	const fullRepoRegex = /([a-zA-Z0-9._-]+)\/([a-zA-Z0-9._-]+)#(\d+)/g;
```

Comment:
We have a regex for this already:

https://github.com/microsoft/vscode-pull-request-github/blob/e5ce10696b0010b836f8a9a96a4ecd6a33295a48/src/github/utils.ts#L58-L59